docs/source/developers/guide/introduction.rst

```diff
@@ -0,0 +1,142 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+
+.. SCOPE OF THIS SECTION
+.. Make an introduction to the Guide for New Contributors.
+.. Add an introduction to the project, why to get involved
+.. and a quick checklist of the contributing process.
+.. Add the Table of Contents. All detailed explanations
+.. should be added as a link in the content.
+
+
+.. _introduction:
+
+***********************
+New Contributor's Guide
+***********************
+
+This guide is meant to be a resource for contributing to
+`Arrow <https://arrow.apache.org//>`_ for new contributors.
+
+Starting to contribute to a project like Apache Arrow can look
+scary. Taking small steps will help making this tack easier.
+
+
+Why contribute to Arrow?
+========================
+There can be various reasons why someone would want to contribute
+to Arrow:
+
+* You find the project interesting and would like to try making
+  a contribution to learn more about the library and grow your skills.
+
+* You use Arrow in the project you are working on and you would like
+  to make a new feature or fix a bug yourself.
+
+Read more about the project in the :ref:`about_arrow` section.
+
+Quick Reference
+===============
+
+Here are the basic steps needed to get set up and contribute to Arrow.
+This is meant as a checklist and also to have an overall picture.
+
+For complete instructions please follow :ref:`step_by_step` (a
+step-by-step guide) or R and Python :ref:`tutorial-index` for an example
+of adding a basic feature.
+
+
+#. **Install and set up Git, fork the Arrow repository**
+
+   See detailed instructions on how to :ref:`set-up` Git and fork the
+   Arrow repository.
+
+#. **Build Arrow**
+
+   This can be quite easy or a real blocker. Be prepared to dig through the
+   additional flags for cmake in the C++ build guide and read the doc carefully
+   to not miss any suggestions already given there. Anytime you are stuck,
+   reach out to us via appropriate :ref:`communication` channel.
+
+   See detailed instructions on how to :ref:`build-arrow`.
```

Comment:
One general question: is this guide mostly for contributing to the R, Python and C++ parts? When it is the case, it should probably be stated at the beginning of the relevant section rather than remain implicit.

(there are also other implementations in the repository: Java, C#, Go...)

Reply:
It is not meant only for R or Python but we are currently focusing on these two as we have the knowledge for it. Afterwords it would be great to add other languages also.

I could change the text to: "See a short description about the building process of PyArrow or R-Arrow (from the guide) or go straight to detailed instructions on how to **build one of Arrow libraries** in the documentation (https://arrow.apache.org/docs/index.html)." - what do you think?